tasks/src/main/scala/dagr/tasks/picard/IlluminaBasecallsToSam.scala

```diff
@@ -0,0 +1,96 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package dagr.tasks.picard
+
+import java.text.SimpleDateFormat
+
+import dagr.core.execsystem._
+import dagr.core.tasksystem.{Retry, VariableResources}
+import dagr.tasks.DagrDef.{DirPath, FilePath}
+import htsjdk.samtools.util.Iso8601Date
+import picard.util.IlluminaUtil.IlluminaAdapterPair
+
+import scala.collection.mutable.ListBuffer
+
+class IlluminaBasecallsToSam(basecallsDir: DirPath,
+                             lane: Int,
+                             runBarcode: String,
+                             readStructure: String,
+                             libraryParamsFile: FilePath,
+                             runDate: Option[Iso8601Date] = None,
+                             sequencingCenter: Option[String] = None,
+                             includeNonPfReads: Boolean = false,
+                             minThreads: Int = 4,
+                             maxThreads: Int = 16,
+                             adapterPairs: Seq[IlluminaAdapterPair] = Seq(
+                               IlluminaAdapterPair.INDEXED,
+                               IlluminaAdapterPair.DUAL_INDEXED,
+                               IlluminaAdapterPair.NEXTERA_V2,
+                               IlluminaAdapterPair.FLUIDIGM
+                             ),
+                             barcodesDir: Option[DirPath] = None,
+                             maxReadsInRamPerTile: Option[Int] = Some(500000),
+                             tmpDir: Option[DirPath] = None
+                            ) extends PicardTask with VariableResources with Retry {
+
+  protected val byMemoryPerThread: Memory = Memory("1GB")
+  protected var memoryPerThread: Memory = Memory("2GB")
+
+  /** Increases the memory per core/thread and returns true if we can run with the fewest # of threads. */
+  override def retry(systemResources: SystemResources, taskInfo: TaskExecutionInfo): Boolean = {
```

Comment:
I wonder if you should mix in one or more of the OOM traits so that this retry only happens if an OOM was detected and not some other kind of error?